script/run_query

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+"""
+Runs SQL queries and writes results to destination tables.
+
+When executing a query associated metadata is parsed to determine whether
+results should be written to a corresponding public dataset.
+"""
+
+from argparse import ArgumentParser
+import os
+import re
+import subprocess
+import sys
+import yaml
+
+
+# sys.path needs to be modified to enable package imports from parent
+# and sibling directories. Also see:
+# https://stackoverflow.com/questions/6323860/sibling-package-imports/23542795#23542795
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from bigquery_etl.parse_metadata import Metadata  # noqa E402
+
+
+PROJECT_ID_ARG_RE = re.compile(fr"(--destination_table=)(\s?[a-zA-z0-9_-]+)")
+
+parser = ArgumentParser(description=__doc__)
+parser.add_argument(
+    "--public_project_id",
+    default="mozilla-public-data",
+    help="Project with publicly accessible data",
+)
+parser.add_argument(
+    "--destination_table", help="Destination table name results are written to"
+)
+parser.add_argument("--dataset_id", help="Destination dataset")
+parser.add_argument("--query_file", help="File path to query to be executed")
+
+
+def main():
+    args, query_arguments = parser.parse_known_args()
+    query_file = args.query_file
+
+    if args.dataset_id is not None:
+        # dataset ID was parsed by argparse but needs to be passed as parameter
+        # when running the query
+        query_arguments.append("--dataset_id={}".format(args.dataset_id))
+
+    use_public_table = False
+
+    try:
+        metadata = Metadata.of_sql_file(query_file)
+        if metadata.is_public_bigquery():
+            # change the destination table to write results to the public dataset
+            # a view to the public table in the internal dataset is created
+            # when CI runs
+            if args.dataset_id is not None and args.destination_table is not None:
```

Comment:
I'm worried this logic makes too many assumptions. It's possible to run `bq query` with no dataset specified, and with destionation_table set to something like `my_dataset.my_table`. Perhaps we should throw an error here if the options look like they don't match this particular format.

Reply:
I added a regex for the destination table for more validation